### DISCUSSION TITLE
How to turn off all auto import functions except "browser"

### DISCUSSION THREAD

**Discussion by aiktb** [wxt-dev/wxt#1309] [Q&A]
I hate auto-imports, it often makes the code messy and hard to refactor, so I want to delete it.

But when I use `imports: false`, `browser.*` also disappears, so I have to import `browser`.

I use `browser.*` as a replacement for  global `chrome.*`, I don't want to import them in the code anyway, and the only thing I really need in imports.d.ts is `browser`.

**Comment by aklinker1** [comment]
You can use the `config:resolved` hook remove all the default values except for `browser`.

**Reply by aiktb** [reply]
> You can use the `config:resolved` hook remove all the default values except for `browser`.


```ts
    "config:resolved": (wxt) => {
      if (wxt.config.imports) {
        wxt.config.imports = {
          presets: [{ package: "wxt/browser/chrome" }],
          eslintrc: wxt.config.imports.eslintrc,
        };
      }
      console.log(wxt.config.imports);
    },
```

This didn't work, even though the console output only had the following, all the built-in types were still generated.

```js
{
  presets: [ { package: 'wxt/browser/chrome' } ],
  eslintrc: {
    enabled: false,
    filePath: '/home/aiktb/Github/FuriganaMaker/packages/extension/.wxt/eslintrc-auto-import.json',
    globalsPropValue: true
  }
}
```

**Comment by garretwilson** [comment]
What's the latest status on this? I'm reading all the WXT documentation before trying to use it, and the first thing I wanted to do was to disable auto-imports. But if that makes `browser.*` disappear, that suddenly means I have to jump through _more_ hoops developing with WXT just to get `browser.*` back, than if I were to develop a native extension. The workaround looks verbose and confusing without an intimate knowledge of everything under the hood.

So the workaround we should use is the first block in [the comment above](https://github.com/wxt-dev/wxt/discussions/1309#discussioncomment-11681510)?
